src/main/java/org/apache/ibatis/migration/options/Options.java

```diff
@@ -15,6 +15,85 @@
  */
 package org.apache.ibatis.migration.options;
 
+import java.io.File;
+
 public enum Options {
-  PATH, ENVPATH, SCRIPTPATH, DRIVERPATH, HOOKPATH, ENV, FORCE, TRACE, HELP, TEMPLATE, IDPATTERN
+
+  PATH {
+    @Override
+    public void accept(SelectedOptions options, String arg) {
+      options.getPaths().setBasePath(new File(arg));
+    }
+  },
+  ENVPATH {
+    @Override
+    public void accept(SelectedOptions options, String arg) {
+      options.getPaths().setEnvPath(new File(arg));
+    }
+  },
+  SCRIPTPATH {
+    @Override
+    public void accept(SelectedOptions options, String arg) {
+      options.getPaths().setScriptPath(new File(arg));
+    }
+  },
+  DRIVERPATH {
+    @Override
+    public void accept(SelectedOptions options, String arg) {
+      options.getPaths().setDriverPath(new File(arg));
+    }
+  },
+  HOOKPATH {
+    @Override
+    public void accept(SelectedOptions options, String arg) {
+      options.getPaths().setHookPath(new File(arg));
+    }
+  },
+  ENV {
+    @Override
+    public void accept(SelectedOptions options, String arg) {
+      options.setEnvironment(arg);
+    }
+  },
+  FORCE {
+    @Override
+    public void accept(SelectedOptions options, String ignored) {
+      options.setForce(true);
+    }
+  },
+  TRACE {
+    @Override
+    public void accept(SelectedOptions options, String ignored) {
+      options.setTrace(true);
+    }
+  },
+  HELP {
+    @Override
+    /**
+     * @ignored could used used to get help on a certain topic if implemented
+     */
+    public void accept(SelectedOptions options, String ignored) {
+      options.setHelp(true);
+    }
+  },
+  TEMPLATE {
+    @Override
+    public void accept(SelectedOptions options, String arg) {
+      options.setTemplate(arg);
+    }
+  },
+  IDPATTERN {
+    @Override
+    public void accept(SelectedOptions options, String arg) {
+      options.setIdPattern(arg);
+    }
+  },
+  QUIET {
+    @Override
+    public void accept(SelectedOptions options, String arg) {
+      options.setQuiet(true);
+    }
+  };
+  public abstract void accept(SelectedOptions options, String arg);
```

Comment:
This also happens. Not everything needs an arg.

Reply:
True - with strongly typed languages you can't do like javascript does and just get magical arguments trailing on the end. with JS everything is a vararg and you use as many args as you want. Kotlin has a nice feature whereby you can default values. Java isn't like that :(

We can create an overload that does 

```java
    public void accept(SelectedOptions option) { 
        accept(options,null);
    }
```

No big deal